src/commands/dev-env-sync-sql.ts

```diff
@@ -27,8 +47,12 @@ import { getReadInterface } from '../lib/validations/line-by-line';
 function findSiteHomeUrl( sql: string ): string | null {
 	const regex = `['"](siteurl|home)['"],\\s?['"](.*?)['"]`;
 	const url = sql.match( regex )?.[ 2 ] || '';
-
-	return urlLib.parse( url ).hostname || null;
+	try {
+		new URL( url );
+		return url;
+	} catch {
+		return null;
+	}
```

Comment:
Node 18.17.0 has [`URL.canParse()`](https://nodejs.org/api/url.html#urlcanparseinput-base); hopefully, someday we can use it instead of this code.